File: Data Structures & Algorithms/house-robber-ii/submission-0.cpp
class Solution {
    private:
    int helper(vector<int> nums){
        int n = nums.size();
        vector<int> dp(n+2);
        dp[1] = nums[0];
        for(int i = 2;i<=n;i++){
            dp[i] = max(dp[i-1],dp[i-2] + nums[(n+ i-1)%n]);
        }
        return dp[n];
    }
public:
    int rob(vector<int>& nums) {
        int n = nums.size();
        if(n == 1) return nums[0];
        return max(helper(vector<int>(nums.begin() + 1, nums.end())),
                helper(vector<int>(nums.begin(), nums.end() - 1)));
    }
};

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-0.cs
public class Solution {
    public int FindKthLargest(int[] nums, int k) {
        PriorityQueue<int,int> maxHeap = new PriorityQueue<int,int>(
            Comparer<int>.Create((x,y) => y.CompareTo(x))
        );
        foreach(int num in nums){
            maxHeap.Enqueue(num,num);
        }

        for(int i=1;i<k;i++){
            maxHeap.Dequeue();
        }
        return maxHeap.Dequeue();
    }
}
